gregBorchers/HW5Problem3.js

```diff
@@ -0,0 +1,164 @@
+//---- make User with closure -----
+function makeUserFactory() {
+
+    var userMsgLog = []; // persists through all calls to the factory()
+
+    function makeUser(newUserName, newPassword) {
+
+        var user = {
+            // "en-closured" methods
+            username: function() {
+                return newUserName;
+            },
+            pwd: function() {
+                return newPassword;
+            },
```

Comment:
Because these two functions are included in `user`, which you are returning, they serve to expose the information that you want to control access to. The two local variables `newUserName` and `newPassword` would normally be locked away inside the closure, but providing these two getter methods will allow the values to be seen.